src/planwise/component/scenarios.clj

```diff
@@ -69,14 +70,23 @@
     (seq (map select-fn providers))))
 
 (defn get-scenario-for-project
-  [store scenario {:keys [provider-set-id provider-set-version config] :as project}]
+  [store scenario {:keys [provider-set-id provider-set-version config source-set-id] :as project}]
   (let [filter-options (-> (select-keys project [:region-id :coverage-algorithm])
                            (assoc :tags (get-in config [:providers :tags])
                                   :coverage-options (get-in config [:coverage :filter-options])))
-        initial-providers  (get-initial-providers store provider-set-id provider-set-version filter-options)]
+        initial-providers  (get-initial-providers store provider-set-id provider-set-version filter-options)
+        sources-data       (edn/read-string (:sources-data scenario))
```

Comment:
Serialization format is an implementation detail which should be moved into a lower "layer". Eg. deserialize in `get-scenario`.